lib/lhm/retry_helper.rb

```diff
@@ -0,0 +1,43 @@
+require 'retriable'
+
+module Lhm
+  # RetryHelper standardizes the interface for retry behavior in components like
+  # Entangler and AtomicSwitcher.
+  #
+  # It expects the caller to implement:
+  # DEFAULT_MAX_RETRIES
+  # DEFAULT_RETRY_WAIT
+  #
+  # These defaults can be overridden by setting an instance variable with corresponding names:
+  # @max_retries
+  # @retry_wait
+  #
+  # To retry some behavior:
+  # Retriable.retriable(retry_config) do
+  #   @connection.execute(tagged(stmt))
+  # end
+  module RetryHelper
+    def self.included(base)
+      raise(ArgumentError, "#{base} must define DEFAULT_MAX_RETRIES before calling 'include RetryHelper'") unless base.constants.include?(:DEFAULT_MAX_RETRIES)
+      raise(ArgumentError, "#{base} must define DEFAULT_RETRY_WAIT before calling 'include RetryHelper'") unless base.constants.include?(:DEFAULT_RETRY_WAIT)
+    end
+
+    def retry_config
+      {
+        on: {
+          StandardError => [/Lock wait timeout exceeded/]
```

Comment:
I know this is what the original code said, but I think we can assume `Mysql2::Error` to be a little more specific.